http/client.go

```diff
@@ -479,6 +479,16 @@ func (c *Client) Purge(ctx context.Context) error {
 	return err
 }
 
+func (c *Client) HealthCheck(ctx context.Context) error {
```

Comment:
question: This appears to be unrelated, is it?

Reply:
It's not. In order to ensure that the `node.Start` method returns only when the server is fully started, we need to make a call to the server. Basically what happens when starting the server is that we instantiate the listener which secures the port number and allow clients to connect. The `server.Serve()` method is the one that enables handling the clients' requests and that can take a bit of time to setup and is done in a goroutine because it blocks until the server is closed. So to ensure that `server.Serve()` is all good to go, we send a `HealthCheck` request that returns only when the server is truly operational.

Doing this ensure that subsequent operations will behave as expected.